services/spar/src/Spar/Scim/User.hs

```diff
@@ -235,13 +235,15 @@ validateScimUser' idp richInfoLimit user = do
     -- Validate rich info (@richInfo@). It must not exceed the rich info limit.
     validateRichInfo :: RichInfo -> m RichInfo
     validateRichInfo richInfo = do
-        let size = richInfoSize richInfo
-        when (size > richInfoLimit) $ throwError $
-            (Scim.badRequest Scim.InvalidValue
-                 (Just . cs $
-                      "richInfo exceeds the limit: max " <> show richInfoLimit <>
-                      " characters, but got " <> show size))
-            { Scim.status = Scim.Status 413 }
+        let errorIfTooBig s name =
+              when (s > richInfoLimit) $ throwError $
+              (Scim.badRequest Scim.InvalidValue
+                (Just . cs $
+                  cs name <> " exceeds the limit: max " <> show richInfoLimit <>
+                  " characters, but got " <> show s))
+              { Scim.status = Scim.Status 413 }
+        errorIfTooBig (richInfoAssocListSize $ richInfoAssocList richInfo) richInfoAssocListURN
+        errorIfTooBig (richInfoMapSize richInfo) richInfoMapURN
```

Comment:
I think this doubles the effective rich info limit.  Why don't you just add the two sizes?  But I don't think it's a problem, we put some pretty ad-hoc number here anyway.

Reply:
I thought eventually people would be moving stuff from old rich info to new one, but we'll continue storing both. So I didn't want to break stuff just because now there is more data and the limit is same.